### PR TITLE
RyuJIT/x86: convert SBCG to NYI for multi-level expression trees

### PR DESCRIPTION
**Pull request by BruceForstall**
In general, we decompose longs by walking the tree bottom-up. So, e.g.,
we'll decompose 'add' as follows:
```
  + (gt_long(lo1, hi1), gt_long(lo2, hi2))
=>
  gt_long (+ (lo1, lo2), +Hi (hi1, hi2))
```
But what if lo1 is '+' and hi1 is +Hi? So,
```
  + (gt_long( + (lo3, lo4), +Hi (hi3, hi4) ), gt_long(lo2, hi2))
=>
  gt_long (+ ( + (lo3, lo4), lo2), +Hi ( +Hi (hi3, hi4), hi2))
```
But we can't put any instructions between `+ (lo3, lo4)` and `+Hi (hi3, hi4)`
or we'll lose the carry bit.

We need to model this differently, or introduce temps. Until this is
done, convert the silent bad codegen into NYI fallback.

### CODE REVIEW

**Comment by BruceForstall** [comment]
@CarolEidt @adiaaida PTAL
/cc @dotnet/jit-contrib 

This adds about 24 new NYI fallbacks to the OSS test run.

**Comment by adiaaida** [comment]
LGTM

**Comment by CarolEidt** [comment]
LGTM

File: src/jit/lower.cpp [hunk 1]
```diff
@@ -428,6 +428,27 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
...
+            // they are decomposed nodes), then we also can't decompose the node, as we aren't
+            // guaranteed the high and low parts will be executed immediately after each other.
+            
+            NYI_IF(hiOp1->OperIsHigh() ||
+                   hiOp2->OperIsHigh() ||
+                   (GenTree::OperIsHigh(getHiOper(oper)) &&
+                    (!loOp1->IsLocal() ||
+                     !hiOp1->IsLocal() ||
+                     !loOp1->IsLocal() ||
+                     !hiOp2->IsLocal())),
```
Comment (sivarv) [inline src/jit/lower.cpp]: if the second operand is an immediate value (or const), we don't need to NYI?

Comment (BruceForstall) [inline src/jit/lower.cpp]: I don't think so. But we'll find out :-)

**Comment by BruceForstall** [comment]
@mmitche It looks like it took almost 1 hour 20 minutes between scheduling the OSX test run and actually starting it:
```
11:02:47 Schedule job dotnet_coreclr » master » checked_osx_tst_prtest
12:20:01 Build dotnet_coreclr » master » checked_osx_tst_prtest #1296 started
```
Is that because there aren't enough Macs, and it was waiting for one?

**Comment by mmitche** [comment]
@BruceForstall Yes.

**Comment by BruceForstall** [comment]
@dotnet-bot test this please

**Comment by BruceForstall** [comment]
@dotnet-bot test Windows_NT x86ryujit

**Pull request merged**